Added Commands for Lights

avakimc opened the pull request:
- Added buttons on operatorController to toggle lights for CONE and CUBE signaling
- Added instantCommands that toggle the lights during prepScore and finishScore to indicate the status of the robot
- For all of these methods, we added lights as a parameter

avakimc committed:
Added Commands for Lights

- Added buttons on operatorController to toggle lights for CONE and CUBE signaling
- Added instantCommands that toggle the lights during prepScore and finishScore to indicate the status of the robot
- For all of these methods, we added lights as a parameter

avakimc committed:
Created togglePartyMode()

* mapped right d-pad on operatorController to togglePartyMode()
* partyMode switches from colors in the rainbow
* added new LightCode that is just the color orange to have a full rainbow for partyMode (per James's request)

tails618 commented on src/main/java/frc/robot/commands/TwoGamePiecesThatEngage.java:
```diff
@@ -67,27 +69,30 @@ public TwoGamePiecesThatEngage(
-        /** TODO: Limelight code goes here */
-        new InstantCommand(lift::ManualPrepScoreSequence, lift),
+        /** TODO: Limelight code goes here 
+         * if limelight fails to find valid target/arpil tag, then turn on NO_TAG light
```
I'd add this now. The logic is going to be *very* simple, since the LL has a boolean function for if it sees a target. It shouldn't take more than a few minutes.

avakimc committed:
Revised Lights Controls to Use Blinkin

- removed rainbow code and replaced it with the built-in blinkin rainbow code
- unsure of how to change spacing when using blinkin, if even necessary

tails618 commented on src/main/java/frc/robot/subsystems/Lights.java:
```diff
@@ -23,41 +21,29 @@ public class Lights extends SubsystemBase {
     NO_TAG, // Blinking Green
     WORKING, // Solid Red
     READY_TO_SCORE, // Blue
+    ORANGE,
```
Remove

voidSevenSevenSix committed:
Deleted orange, added code for if limelight doesn't see a tag

tails618 commented on src/main/java/frc/robot/commands/TwoGamePiecesThatEngage.java:
```diff
@@ -85,9 +86,8 @@ public TwoGamePiecesThatEngage(
-         * if limelight fails to find valid target/arpil tag, then turn on NO_TAG light
-        */
+        /** TODO: Limelight code goes here */
+        new InstantCommand(()->{if(tagLimelight.getValidTarget() == 0){lights.toggleCode(LightCode.NO_TAG);}}),
```
Instead use `isValidTarget()` which returns a boolean (line 220 in TagLimelight)

voidSevenSevenSix committed:
Changed getValidTarget==0 to !isvalidtarget

voidSevenSevenSix commented:
/format

github-actions[bot] committed:
Formatting fixes
